scripts/check_overall_backend_test_coverage_test.py

```diff
@@ -0,0 +1,158 @@
+
+        with swap_subprocess_run, self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            RuntimeError,
+            'Run backend tests before running this script. ' +
+            '\nOUTPUT: No data to report.\nERROR: None'
+        ):
+            check_overall_backend_test_coverage.main()
+
+    def test_failure_to_execute_coverage_command_throws_error(self) -> None:
+        class MockProcess:
+            returncode = 1
+            stdout = 'Some error occured.'
+            stderr = 'Some error.'
+        def mock_subprocess_run(*args: Any, **kwargs: Any) -> MockProcess: # pylint: disable=unused-argument
+            return MockProcess()
+        swap_subprocess_run = self.swap_with_checks(
+            subprocess, 'run', mock_subprocess_run,
+            expected_args=((self.cmd,),),
+            expected_kwargs=[{
+                'capture_output': True,
+                'encoding': 'utf-8',
+                'env': self.env,
+                'check': False
+            }])
+
+        with swap_subprocess_run, self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            RuntimeError,
+            'Failed to calculate coverage because subprocess failed. ' +
+            '\nOUTPUT: Some error occured.\nERROR: Some error.'
+        ):
+            check_overall_backend_test_coverage.main()
+
+    def test_error_in_parsing_coverage_report_throws_error(self) -> None:
+        class MockProcess:
+            returncode = 0
+            stdout = 'TOTALL     40571  10682  13759   1161   70% '
+        def mock_subprocess_run(*args: Any, **kwargs: Any) -> MockProcess: # pylint: disable=unused-argument
+            return MockProcess()
+        swap_subprocess_run = self.swap_with_checks(
+            subprocess, 'run', mock_subprocess_run,
+            expected_args=((self.cmd,),),
+            expected_kwargs=[{
+                'capture_output': True,
+                'encoding': 'utf-8',
+                'env': self.env,
+                'check': False
+            }])
+
+        with swap_subprocess_run, self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            RuntimeError, 'Error in parsing coverage report.'
+        ):
+            check_overall_backend_test_coverage.main()
+
+    def test_overall_backend_coverage_checks_failed(self) -> None:
+        class MockProcess:
+            returncode = 0
+            stdout = 'TOTAL     40571  10682  13759   1161   70% '
+        def mock_subprocess_run(*args: Any, **kwargs: Any) -> MockProcess: # pylint: disable=unused-argument
+            return MockProcess()
+        swap_subprocess_run = self.swap_with_checks(
+            subprocess, 'run', mock_subprocess_run,
+            expected_args=((self.cmd,),),
+            expected_kwargs=[{
+                'capture_output': True,
+                'encoding': 'utf-8',
+                'env': self.env,
+                'check': False
+            }])
+        swap_sys_exit = self.swap(sys, 'exit', lambda _: None)
```

Comment:
Can you use `self.swap_with_checks` here too so that we make sure the exit code is nonzero?

Reply:
Done!